main.tf

```diff
@@ -144,20 +144,12 @@ resource "github_repository_collaborator" "collaborator" {
 #
 # Repository teams
 #
-locals {
-  team_admin = { for i in var.admin_team_ids : i => "admin" }
-  team_push  = { for i in var.push_team_ids : i => "push" }
-  team_pull  = { for i in var.pull_team_ids : i => "pull" }
-
-  teams = merge(local.team_admin, local.team_push, local.team_pull)
-}
-
 resource "github_team_repository" "team_repository" {
-  for_each = local.teams
+  count = length(var.teams)
```

Comment:
We run into the same issue we had with lists before here no? we can't dynamically add or remove elements inside the list?

Reply:
doe snot really matter for team repositories .. but yes will adjust this.. for easier plan review.... this came from the revert...

Reply:
fixed

Reply:
then it depends on an unknwon variable... the reason why we had to revert it in the first place... ;) so count is the only way possible as of today.... ;) but it does not matter as teams will be added and recreated... which is fine... as this is a 100% automated action without the need to confirm manually for the endusers in github..